mlflow/pyfunc/__init__.py

```diff
@@ -201,34 +201,70 @@ def spark_udf(spark, path, run_id=None, result_type="double"):
     Parameters passed to the UDF are forwarded to the model as a DataFrame where the names are
     ordinals (0, 1, ...).
 
+    If the model returns predictions as DataFrame, different result may be returned based on the
+    requested result_type. If the requested result type is numeric, the data frame is filtered to
+    contain only numeric columns first. If the requested return type is not an ArrayType, only the
+    first column will be returned.
+
     >>> predict = mlflow.pyfunc.spark_udf(spark, "/my/local/model")
     >>> df.withColumn("prediction", predict("name", "age")).show()
 
     :param spark: A SparkSession object.
     :param path: A path containing a :py:mod:`mlflow.pyfunc` model.
     :param run_id: ID of the run that produced this model. If provided, ``run_id`` is used to
                    retrieve the model logged with MLflow.
+    :param result_type: the return type of the user-defined function. The value can be either a
+                        :class:`pyspark.sql.types.DataType` object or a DDL-formatted type string.
```

Comment:
Can we enumerate the classes of acceptable result types and their corresponding UDF behavior? For example:

```
:param result_type: the return type of the user-defined function. The value can be either a
                    :class:`pyspark.sql.types.DataType` object or a DDL-formatted type string.
                    The following classes of result type are supported:
                           
                        - NumericType: The model must produce numeric output in order to use 
                                       this result type.
                        - Array[NumericType]: If the model produces predictions as a DataFrame, 
                                              ``spark_udf`` will filter the DataFrame to remove 
                                               non-numeric columns before returning.
                        - StringType: ...
                        - Array[StringType]: ...
```